tool/generator/lib/src/projection/winrt/declarations/comobject.dart

```diff
@@ -0,0 +1,31 @@
+import '../../../../generator.dart';
+
+class COMObjectMethod extends WinRTMethodProjection {
+  COMObjectMethod(super.method, super.vtableOffset);
+
+  @override
+  String toString() => '''
+      Pointer<COMObject> $camelCasedName($methodParams) {
+        final retValuePtr = calloc<COMObject>();
+        $parametersPreamble
+        ${ffiCall()}
+        $parametersPostamble
+        return retValuePtr;
+      }
+''';
+}
+
+class COMObjectGetProperty extends WinRTGetPropertyProjection {
```

Comment:
Similar to above, confusion between this and `ComGetPropertyProjection`. This would perhaps be `WinRTGetPropertyReturningComObjectProjection`?